pymeasure/instruments/tdk/tdk_base.py

```diff
@@ -0,0 +1,387 @@
+
+        Returns the software version as an ASCII string.
+        """
+    )
+
+    serial = Instrument.measurement(
+        "SN?",
+        """Get the serial number of the instrument.
+
+        Returns the serial number of of the instrument as an ASCII string.
+        """
+    )
+
+    last_test_date = Instrument.measurement(
+        "DATE?",
+        """Get the date of the last test, possibly calibration date.
+
+        Returns a string in the format: yyyy/mm/dd.
+        """
+    )
+
+    voltage = Instrument.control(
+        "PV?", "PV %g",
+        """Control the programmed (set) output voltage.""",
+        validator=lambda v, vs: strict_discrete_range(v, vs, step=0.01),
+        values=[0, 40],
+        dynamic=True
+    )
+
+    actual_voltage = Instrument.measurement(
+        "MV?",
+        """Measure the the actual output voltage."""
+    )
+
+    current = Instrument.control(
+        "PC?", "PC %g",
+        """Control the programmed (set) output current.""",
+        validator=lambda v, vs: strict_discrete_range(v, vs, step=0.01),
+        values=[0, 38],
+        dynamic=True
+    )
+
+    actual_current = Instrument.measurement(
+        "MC?",
+        """Measure the actual output current.
+
+        Returns a float with five digits of precision.
+        """
+    )
+
+    mode = Instrument.measurement(
+        "MODE?",
+        """Measure the output mode of the power supply.
+
+        When power supply is on, the returned value will be either ``'CV'`` for
+        control voltage or ``'CC'`` for or control current. If the power supply
+        is off, the returned value will be ``'OFF'``.
+        """
+    )
+
+    display = Instrument.measurement(
+        "DVC?",
+        """Get the displayed voltage and current.
+
+        Returns a list of floating point numbers in the order of [ measured voltage,
+        programmed voltage, measured current, programmed current, over voltage set point,
+        under voltage set point ].
+        """
+    )
+
+    status = Instrument.measurement(
+        "STT?",
+        """Get the power supply status.
+
+        Returns a list in the order of [ actual voltage
+        (MV), the programmed voltage (PV), the actual current (MC), the
+        programmed current (PC), the status register (SR), and the fault
+        register (FR) ].
+        """
+    )
+
+    pass_filter = Instrument.control(
+        "FILTER?", "FILTER %d",
+        """Control the low pass filter frequency of the A to D converter
+        for voltage and current measurement.
+
+        Valid frequency values are 18, 23, or 46 Hz. Default value is 18 Hz.
+        """,
+        validator=strict_discrete_set,
+        values=[18, 23, 46]
+    )
+
+    source_output = Instrument.control(
+        "OUT?", "OUT %s",
+        """Control the output of the power supply.
+
+        Valid values are ``'ON'`` and ``'OFF'``.
+        """,
+        validator=strict_discrete_set,
+        values=["ON", "OFF"]
+    )
```

Comment:
Why don't you turn it into boolean?

Regarding the naming, please have a look into https://pymeasure.readthedocs.io/en/latest/dev/adding_instruments/instrument.html#your-instrument-s-user-interface

So output_enabled would be a better name.

Reply:
Updated in last commit. I moved to boolean, thanks for suggestion.